src/Workspaces/Core/Portable/Shared/Utilities/AsyncBatchingWorkQueue`2.cs

```diff
@@ -19,7 +19,7 @@ namespace Roslyn.Utilities
     /// along to be worked on.  Rounds of processing happen serially, only starting up after a
     /// previous round has completed.
     /// </summary>
-    internal class AsyncBatchingWorkQueue<TItem>
+    internal class AsyncBatchingWorkQueue<TItem, TResult>
```

Comment:
i wanted this for navbars.  tehre we want to batch up work, but also pass along the model we're producing from that work that other parts can consume.  navbars had to hackily mutate instance state with locks to accomplish this.  this type will allow us to avoid that.